src/texture_override.h

```diff
@@ -0,0 +1,62 @@
+/*
+Minetest
+Copyright (C) 2010-2013 celeron55, Perttu Ahola <celeron55@gmail.com>
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation; either version 2.1 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License along
+with this program; if not, write to the Free Software Foundation, Inc.,
+51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+#pragma once
+
+#include "irrlichttypes.h"
+#include <string>
+#include <vector>
+
+//! Bitmask enum specifying what a texture override should apply to
+enum class OverrideTarget : u8 {
+	INVALID = 0,
+	TOP = 1 << 0,
+	BOTTOM = 1 << 1,
+	LEFT = 1 << 2,
+	RIGHT = 1 << 3,
+	FRONT = 1 << 4,
+	BACK = 1 << 5,
+	INVENTORY = 1 << 6,
+	WIELD = 1 << 7,
+
+	SIDES = LEFT | RIGHT | FRONT | BACK,
+	ALL_FACES = TOP | BOTTOM | SIDES,
+	ITEM_TARGETS = INVENTORY | WIELD,
+};
+
+struct TextureOverride {
+	std::string id;
+	std::string texture;
+	u8 target;
```

Comment:
We can do that but...it's going to severely impact the readability of the code. Unfortunately, `enum class` doesn't take well to bitwise operations so we'd have to insert a bunch of additional `static_cast<>` calls. On the other hand, we could probably use a plain `enum` but it will bleed into the global namespace which isn't really ideal either. Hence, the current situation where I've made this value match the underlying type of the `enum class`.

Taking that into consideration, do you still feel that this is the best call? Or would you prefer that I go for one of the other options?

Reply:
Doesn't turning `OverrideTarget` from an `enum class` into an `enum` fix that?

Reply:
It should, I guess I can just prefix the values.

Reply:
It's weird to me for an enum to have a value that is not an option in the enum

Reply:
My suggestion then:
* In the parsing code use an u8 and cast just once
* Add an inline helper function to `OverrideTarget` so the code in nodedef/itemdef has no casts

Reply:
I *think* I've made the change you wanted. Leaving unresolved so you can confirm.